app/client/ui/CustomWidgetGallery.ts

```diff
@@ -235,6 +238,14 @@ class CustomWidgetGallery extends Disposable {
   private _buildWidget(info: WidgetInfo) {
     const {variant, id, name, description, developer, lastUpdated} = info;
 
+    if (variant === 'custom') {
+      this._customUrlInput = cssCustomUrlInput(
+        this._customUrl,
+        {placeholder: t('Widget URL'), type: 'url'},
+        testId('custom-url'),
+      );
+    }
```

Comment:
Could you explain a bit more about the motivations behind this change? It seems a bit enigmatic to me by reading the changes.

Here is my understanding and my doubts:
1. We check that `variant` equals to `"custom"`, and if that's the case, we create a new `cssCustomUrlInput` no matter if there already existed a value for `this._customUrlInput`;
2. We check again that `variant` equals to `"custom"` below to actually use this value;

Was the intent behind to lazily create this component only when we need it? In that case, I would suggest to rather create a singleton to call line 309:
```diff
- variant !== 'custom' ? null : this._customUrlInput,
+ variant !== 'custom' ? null : this._getCustomUrlInput(), // or whatever function name
```

Reply:
This was done so that the custom url input can easily be targeted outside of the dom building function (in `_validateCustomWidgetUrl`) in order to validate it. This assumes we only have one "custom url widget" rendered in the gallery but I figured this was already assumed in the code so doing it this way was ok. I guess your approach would also work only when assuming that.

I didn't write this to lazy load it. I guess  I could initiate the custom url input directly in the constructor to be more explicit?

Reply:
I did just that (directly initiate the variable in the construtor), it feels more explicit and less weird than changing this global value when requesting to build a dom el. Thanks for the heads up.

Reply:
It would be more explicit, and I guess less error prone.

Also I wonder if there is no room for converting this modal dialog content into a true form, so that we may just check the result of `form.reportValidity()`. I guess that's a work for another day.

Reply:
> Also I wonder if there is no room for converting this modal dialog content into a true form, so that we may just check the result of form.reportValidity(). **I guess that's a work for another day**

Yeah 😅